code/modules/projectiles/gun.dm

```diff
@@ -373,6 +374,13 @@
 	if (!istype(P))
 		return //default behaviour only applies to true projectiles
 
+	var/obj/structure/simple_door/key_door/door = target
```

Comment:
Why did you double the code in another .dm if it was apparently working previously?

Reply:
At the time it wasn't working correctly that's  remeents from coding at 1AM will remove in a bit